lib/puppet/provider/oneview_storage_pool/c7000.rb

```diff
@@ -37,12 +37,15 @@ def create
   end
 
   def manage
+    return unless @client.api_version >= 500
     is_managed = @data.delete('isManaged')
     get_single_resource_instance.manage(is_managed)
     true
   end
 
   def reachable
+    return unless @client.api_version >= 500
+    @data['uri'] = OneviewSDK.resource_named('StoragePool', api_version).find_by(@client, name: @data['uri']).first['uri']
```

Comment:
Give the decision to user whether to pass "name" or "uri". Keep the key names different.
```
temp_name = @data.delete('name')
@data['uri'] = OneviewSDK.resource_named('StoragePool', api_version).find_by(@client, name: temp_name).first['uri']
```